### DISCUSSION TITLE
how to get information about the execution end of functions and return values in run_in_threadpool

### DISCUSSION THREAD

**Discussion by cherepanovic** [fastapi/fastapi#7156] [Questions] [answered]
Hello community,

I am using the run_in_threadpool to execute a function in a different thread 

`await run_in_threadpool(pp.do_epochs, indices)`

but I could not find any information on how to get the return values of the function in run_in_threadpool and how to know when the function is terminated.

Thank you in advance!

**Comment by cherepanovic** [comment]
have to read about asyncio lib, loop.run_in_executor and so on

**Comment by tiangolo** [comment, marked as answer]
Hey there! I created Asyncer to help with that: https://asyncer.tiangolo.com/

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
